backend/src/main/java/agh/toik/controller/QuestionsApiController.java

```diff
@@ -1,55 +1,71 @@
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
 
+import javax.annotation.Generated;
+import javax.validation.constraints.*;
 import javax.validation.Valid;
 import javax.servlet.http.HttpServletRequest;
-import java.io.IOException;
+import java.util.List;
 
-@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.SpringCodegen", date = "2020-05-19T18:19:31.581Z[GMT]")
+@Generated(value = "io.swagger.codegen.v3.generators.java.SpringCodegen", date = "2020-05-26T17:58:06.365Z[GMT]")
 @Controller
 public class QuestionsApiController implements QuestionsApi {
 
     private static final Logger log = LoggerFactory.getLogger(QuestionsApiController.class);
-
     private final ObjectMapper objectMapper;
-
     private final HttpServletRequest request;
 
     private QuestionRepo questionRepo;
 
-    @org.springframework.beans.factory.annotation.Autowired
+    @Autowired
     public QuestionsApiController(ObjectMapper objectMapper, HttpServletRequest request, QuestionRepo questionRepo) {
         this.objectMapper = objectMapper;
         this.request = request;
         this.questionRepo = questionRepo;
     }
 
-    public ResponseEntity<Question> addQuestion(@ApiParam(value = "Question object that needs to be added to the db" ,required=true )  @Valid @RequestBody Question entity
-) {
+    public ResponseEntity<Question> addQuestion(@Valid @RequestBody Question entity) {
         entity.setId(Helper.generateID());
         questionRepo.save(entity);
         return ResponseEntity.ok(entity);
     }
 
-    public ResponseEntity<Question> updateQuestion(@ApiParam(value = "Question object that needs to be updated to the db" ,required=true )  @Valid @RequestBody Question entity
-) {
+    public ResponseEntity<Void> deleteQuestion(@PathVariable("questionId") Long questionId) {
+        Question questionToDelete = questionRepo.findOne(questionId);
+        if (questionToDelete == null) {
+            return new ResponseEntity<Void>(HttpStatus.NOT_FOUND);
+        }
+
+        questionRepo.delete(questionToDelete);
+        return new ResponseEntity<Void>(HttpStatus.NO_CONTENT);
+    }
+
+    public ResponseEntity<List<Question>> getQuestions(@NotNull @Valid @RequestParam(value = "category", required = true) Long category) {
+        List<Question> questions = questionRepo.findQuestionsByCategory(category);
+        return ResponseEntity.ok(questions);
+    }
+
+    public ResponseEntity<Question> updateQuestion(@Valid @RequestBody Question entity) {
         Question questionToUpdate = questionRepo.findOne(entity.getId());
         if (questionToUpdate == null) {
             return new ResponseEntity<Question>(HttpStatus.NOT_FOUND);
         }
 
-        questionRepo.save(entity); //save updates object that has _id value
+        questionRepo.save(entity); // updates object that has _id value
```

Comment:
Missing checking if provided category exists; to match API should be returned 405

It's possible to pass incorrect info about category (ex. witch changed name)
I suggest during checking if category exists, replace it with returned result (or change api.yaml, that Question contains categoryId only)